pmlc/dialect/pxa/transforms/vectorize.cc

```diff
@@ -158,11 +198,27 @@ class Impl {
           builder.create<vector::BroadcastOp>(op.getLoc(), vecType, val);
       val = bcast.getResult();
     }
-    auto vecOp = builder.create<PxaVectorReduceOp>(
-        op.getLoc(), ArrayRef<Type>{op.getMemRefType()}, op.agg(), val,
-        op.memref(), op.map(), op.idxs());
-    op.replaceAllUsesWith(vecOp.getResult());
-    op.erase();
+    // Add vector_reduction only if the stride is 0
+    if (zeroStrideReductions.find(op.getOperation()) !=
```

Comment:
I think you can use `zeroStrideReductions.count(op.getOperation())` here instead of find.

Reply:
Yes, it could be... At least in some STL implementations the count() would do also the find internally (well, really a hash lookup, which find boils down to, and then get the count, which is probably a bit slower than the find and compare to a constant. I might be wrong, since this doesn't hold for all implementation of the STL interfaces, but I don't consider it being a big issue. If you insist, I can change it...

Reply:
FYI, they have equal performance and so the decision to use one or the other should be based on semantics rather than runtime performance: https://stackoverflow.com/questions/14159682/unordered-map-which-one-is-faster-find-or-count?rq=1